src/components/vm/overview/watchdog.tsx

```diff
@@ -144,18 +151,15 @@ export const WatchdogModal = ({
                 .finally(() => setInProgress(false));
     };
 
-    const detach = () => {
-        cockpit.assert(vm.watchdog.model);
+    const detach = async () => {
         setInProgress(true);
-        return domainRemoveWatchdog({
-            connectionName: vm.connectionName,
-            vmName: vm.name,
-            hotplug: vm.state === "running",
-            permanent: vm.persistent,
-            model: vm.watchdog.model,
-        })
-                .then(Dialogs.close, exc => setDialogError({ text: _("Failed to detach watchdog"), detail: exc.message, variant: "danger" }))
-                .finally(() => setInProgress(false));
+        try {
+            await virtXmlHotRemove(vm, "watchdog", { model: vm.watchdog.model });
+            Dialogs.close();
+        } catch (exc) {
+            setDialogError({ text: _("Failed to detach watchdog"), detail: String(exc), variant: "danger" });
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2334-68be91e9-20250929-092324-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)